echopype/echodata/echodata.py

```diff
@@ -403,12 +403,29 @@ def update_platform(
             extra_platform_data = extra_platform_data.drop_vars(trajectory_var)
             extra_platform_data = extra_platform_data.swap_dims({"obs": time_dim})
 
-        # TODO: Expand the clipping range by one (or 2?) indices
+        # clip incoming time to 1 less than min of EchoData.beam["ping_time"] and
+        #   1 greater than max of EchoData.beam["ping_time"]
+        # account for unsorted time
+        sorted_external_time = extra_platform_data[time_dim].data
+        sorted_external_time.sort()
+        min_index = max(
+            np.searchsorted(
+                sorted_external_time, self.beam["ping_time"].min(), side="right"
+            )
+            - 1,
```

Comment:
Why use "right" instead of "left" and then -1? 
For example, in the case when you have [1,2,3,3,3,4,5], to find the index to insert a new 3, wouldn't it be side="left" and then -1? Otherwise you get another 3 in the outcome.

Reply:
Also, a small nitpicking thing: could you combine the -1 to the line above it? The code is more readable that way.

Reply:
I think I mixed up left and right